lib/element.ts

```diff
@@ -178,32 +178,28 @@ export class ElementArrayFinder extends WebdriverWebElement {
               }
             });
       } else {
-        return this.getWebElements().then(
-            (parentWebElements: webdriver.WebElement[]) => {
-              // For each parent web element, find their children and construct
-              // a
-              // list of Promise<List<child_web_element>>
-              let childrenPromiseList = parentWebElements.map(
-                  (parentWebElement: webdriver.WebElement) => {
-                    return locator.findElementsOverride ?
-                        locator.findElementsOverride(
-                            ptor.driver, parentWebElement, ptor.rootEl) :
-                        parentWebElement.findElements(locator);
-                  });
+        return this.getWebElements().then((parentWebElements: WebElement[]) => {
+          // For each parent web element, find their children and construct
+          // a list of Promise<List<child_web_element>>
+          let childrenPromiseList = parentWebElements.map(
+              (parentWebElement: webdriver.WebElement) => {
+                return locator.findElementsOverride ?
+                    locator.findElementsOverride(
+                        ptor.driver, parentWebElement, ptor.rootEl) :
+                    parentWebElement.findElements(locator);
+              });
 
-              // Resolve the list of Promise<List<child_web_elements>> and merge
-              // into
-              // a single list
-              return wdpromise.all(childrenPromiseList)
-                  .then((resolved: webdriver.WebElement[]) => {
-                    return resolved.reduce(
-                        (childrenList: webdriver.WebElement[],
-                         resolvedE: webdriver.WebElement) => {
-                          return childrenList.concat(resolvedE);
-                        },
-                        []);
-                  });
-            });
+          // Resolve the list of Promise<List<child_web_elements>> and merge
+          // into a single list
+          return wdpromise.all(childrenPromiseList).then((resolved: any) => {
```

Comment:
Why `any`? `resolved` seems to be `WebElement[][]`. But again I think it's better just to delete the type annotations here. Excessive type annotations are a kind of type assertions, they prevent the compiler from doing its job of finding type errors.